tests/test_block_factory.py

```diff
@@ -230,14 +231,14 @@ def test_xformer_decoder_block(
     input_mask[input_mask < 0.0] = -float("inf")
 
     encoded = encoder_block(inputs)
-    if decoder_block.mha.attention.supports_attention_mask:
+    if decoder_block.supports_attention_mask:
         _ = decoder_block(
             inputs, encoded, encoder_att_mask=att_mask, input_mask=input_mask
         )
 
     # Test different sequence lengths when encoding and decoding
-    if not decoder_block.mha.attention.requires_same_k_q_dimensions:
```

Comment:
this was getting a little obscure, so the encoder and decoder blocks now expose directly these attention flags, I figured it was easier to read and made sense